jib-cli/src/main/java/com/google/cloud/tools/jib/cli/cli2/Jar.java

```diff
@@ -40,22 +38,15 @@
   @SuppressWarnings("NullAway.Init") // initialized by picocli
   private Path jarFile;
 
-  /**
-   * Returns a user configured Path to a jar file.
-   *
-   * @return a path to a jar file
-   */
-  public Path getJarFile() {
-    return jarFile;
-  }
+  private final TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider();
 
   @Override
   public Integer call() {
     globalOptions.validate();
-    Path jarFile = getJarFile();
     try {
```

Comment:
```suggestion
    try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider()) {
```
and remove the field and the `finally` block.

Reply:
oh this is much better